tests/EppoClientTest.php

```diff
@@ -99,215 +49,96 @@ public static function tearDownAfterClass(): void
-                        'name' => 'control',
-                        'value' => 'control',
-                        'typedValue' => 'control',
-                        'shardRange' => [
-                            'start' => 0,
-                            'end' => 50
-                        ]
-                    ],
-                    [
-                        'name' => 'treatment',
-                        'value' => 'treatment',
-                        'typedValue' => 'treatment',
-                        'shardRange' => [
-                            'start' => 50,
-                            'end' => 100
-                        ]
-                    ]
-                ]
-            ]
-        ];
-
-        $experimentConfigRequesterMock = $this->getExperimentConfigurationRequesterMock($mockedResponse);
-        $pollerMock = $this->getPollerMock();
-        $client = EppoClient::createTestClient($experimentConfigRequesterMock, $pollerMock);
-
-        $this->assertNull(
-            $client->getAssignment('subject-10', self::EXPERIMENT_NAME, ['appVersion' => 9])
-        );
-        $this->assertNull(
-            $client->getAssignment('subject-10', self::EXPERIMENT_NAME)
-        );
-        $this->assertEquals(
-            $client->getAssignment('subject-10', self::EXPERIMENT_NAME, ['appVersion' => 11]),
-            'control'
-        );
-    }
 
-    public function testLogsVariationAssignment()
+    /**
+     * @throws GuzzleException
+     * @throws \Psr\SimpleCache\InvalidArgumentException
+     * @throws ClientExceptionInterface
+     */
+    public function skipTestNoGracefulModeThrows()
     {
         $pollerMock = $this->getPollerMock();
-        $mockConfigRequester = $this->getExperimentConfigurationRequesterMock(self::MOCK_EXPERIMENT_CONFIG);
-        $mockLogger = $this->getLoggerMock();
+        $mockConfigRequester = $this->getFlagConfigurationLoaderMock([], new Exception('config loader error'));
+        $mockLogger = $this->getMockBuilder(LoggerInterface::class)->getMock();
 
         $subjectAttributes = [['foo' => 3]];
+        $client = EppoClient::createTestClient($mockConfigRequester, $pollerMock, $mockLogger, false);
 
-        $client = EppoClient::createTestClient($mockConfigRequester, $pollerMock, $mockLogger);
-        $assignment = $client->getAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
-
-        $this->assertEquals('control', $assignment);
+        $this->expectException(Exception::class);
+        $client->getAssignment(self::EXPERIMENT_NAME, 'subject-10', $subjectAttributes);
+        $client->getStringAssignment(self::EXPERIMENT_NAME, 'subject-10', $subjectAttributes);
+        $client->getNumericAssignment(self::EXPERIMENT_NAME, 'subject-10', $subjectAttributes);
+        $client->getBooleanAssignment(self::EXPERIMENT_NAME, 'subject-10', $subjectAttributes);
+        $client->getJSONAssignment(self::EXPERIMENT_NAME, 'subject-10', $subjectAttributes);
```

Comment:
will these other checks ever actually by run?

Reply:
no, actually. they'd be redundant at any rate.